tests/integration/test_api.py

```diff
@@ -383,6 +389,39 @@ def test_annotation_event(client, monkeypatch, httpserver):
     assert result.status_code == 200
 
 
-def test_display_insight_prediction_for_products(client, mocker):
+def test_insight_prediction(client, mocker):
     result = client.simulate_get("/api/v1/insights/predictions")
     assert result.status_code == 200
+    assert result.json == {"count": 0, "predictions": [], "status": "no_predictions"}
+
+    PredictionFactory(barcode="123")
+    result = client.simulate_get("/api/v1/insights/predictions")
+    assert result.status_code == 200
+    data = result.json
+    assert data["count"] == 1
+    assert data["status"] == "found"
+    predictions_data = data["predictions"]
+    assert predictions_data[0]["barcode"] == "123"
+    assert predictions_data[0]["type"] == "category"
+    assert predictions_data[0]["value_tag"] == "en:seeds"
+
+    PredictionFactory(value_tag="en:seeds")
+    result = client.simulate_get("/api/v1/insights/predictions")
+    assert result.status_code == 200
+    data = result.json
+    assert data["count"] == 2
+    assert data["status"] == "found"
+    predictions_data = data["predictions"]
+    assert predictions_data[0]["type"] == "category"
+    assert predictions_data[1]["type"] == "category"
+    assert predictions_data[0]["value_tag"] == "en:seeds"
+    assert predictions_data[1]["value_tag"] == "en:seeds"
+
+
+def test_get_predictions():
```

Comment:
I have added a test case for `get_predictions` as well. But I am unsure if I am covering all the filters you had in mind. 

Let me know if my approach is correct or you expected something else.

Also, 4 other test cases are failing which is weird I think because I did not touch the existing function (wrote new functions and test cases). I did find their reason for failing using PDB. But I think it is best you have a look first lest my fixing makes more mess.